### PR TITLE
CORS-2940: add CI for publish:mixed on azure

### PR DESCRIPTION
**Pull request by jinyunma**
https://issues.redhat.com/browse/CORS-2940

1. Add two CI jobs for publish:Mixed on azure
    - azure-ipi-mixed-apiserver-internal
    - azure-ipi-mixed-ingress-internal
2. Add post check for them to automatic test cases OCP-69090 and OCP-69109

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure

**Comment by jinyunma** [comment]
- workflow cucushift-installer-rehearse-azure-ipi-publish-mixed-apiserver-internal: 

install-config:
```
publish: Mixed
operatorPublishingStrategy:
  apiserver: Internal
  ingress: External
```
passed job link: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/46459/rehearse-46459-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure/1732389966366904320

- workflow cucushift-installer-rehearse-azure-ipi-publish-mixed-ingress-internal:
install-config:
```
publish: Mixed
operatorPublishingStrategy:
  apiserver: External
  ingress: Internal
```
passed job link: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/46459/rehearse-46459-periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure/1732323180149739520

cc @jianlinliu @gpei to review

File: ci-operator/step-registry/ipi/conf/operator-publish-strategy/ipi-conf-operator-publish-strategy-commands.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+set -o errexit
+set -o pipefail
+
+CONFIG="${SHARED_DIR}/install-config.yaml"
+PATCH="/tmp/install-config-mixed-publish.yaml.patch"
+
+cat > "${PATCH}" << EOF
+publish: Mixed
+operatorPublishingStrategy:
+  apiserver: ${APISERVER_PUBLISH_STRATEGY}
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/operator-publish-strategy/ipi-conf-operator-publish-strategy-commands.sh:13]: If `APISERVER_PUBLISH_STRATEGY` is applied with the default value - an empty string, the real publish strategy is still "External" ?

Comment (jianlinliu) [reply ci-operator/step-registry/ipi/conf/operator-publish-strategy/ipi-conf-operator-publish-strategy-commands.sh:13]: Or we can judge `APISERVER_PUBLISH_STRATEGY` or `INGRESS_PUBLISH_STRATEGY`, once non-empty, then set it.

File: ci-operator/step-registry/cucushift/installer/check/azure/publish-mixed/cucushift-installer-check-azure-publish-mixed-commands.sh [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    fi
+
+    echo "(2/4) check that LB rule on port ${API_SERVER_PORT} is created in internal LB..."
+    api_internal_lb_rule=$(cat ${internal_lb_rules} | jq -r ".[] | select(.backendPort==${API_SERVER_PORT})")
+    if [[ -z "${api_internal_lb_rule}" ]]; then
+        echo "ERROR: not found lb rule on port ${API_SERVER_PORT} in internal LB!"
+        check_result=1
+    fi
+
+    echo "(3/4) check that outbound rule is created in public zone..."
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/installer/check/azure/publish-mixed/cucushift-installer-check-azure-publish-mixed-commands.sh:91]: Do we need to check public ip address is configured in public lb?

File: ci-operator/step-registry/ipi/conf/operator-publish-strategy/ipi-conf-operator-publish-strategy-commands.sh (hunk 1 above)
Comment (jinyunma) [reply ci-operator/step-registry/ipi/conf/operator-publish-strategy/ipi-conf-operator-publish-strategy-commands.sh:22]: Fixed.
One of them must be set when publish:Mixed, and if another one is empty string, the default publish strategy is External.

File: ci-operator/step-registry/cucushift/installer/check/azure/publish-mixed/cucushift-installer-check-azure-publish-mixed-commands.sh (hunk 2 above)
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/check/azure/publish-mixed/cucushift-installer-check-azure-publish-mixed-commands.sh]: good point, updated.

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.15-installer-rehearse-azure

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-azure-ipi-mixed-ingress-internal-arm-f14

**Comment by jinyunma** [comment]
cc @liangxia to review

**Pull request merged**